src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -152,71 +152,55 @@ private char ScanEscapeSequence(out char surrogateCharacter)
             return ch;
         }
 
-        private void ScanVerbatimStringLiteral(ref TokenInfo info, bool allowNewlines = true)
+        private ErrorCode? ScanVerbatimStringLiteral(ref TokenInfo info, bool allowNewlines)
         {
             _builder.Length = 0;
 
-            if (TextWindow.PeekChar() == '@' && TextWindow.PeekChar(1) == '"')
+            Debug.Assert(TextWindow.PeekChar() == '@' && TextWindow.PeekChar(1) == '"');
+            TextWindow.AdvanceChar(2);
+
+            ErrorCode? error = null;
+            while (true)
             {
```

Comment:
no need for this 'if' check, the callers ensure it.  so this became a Debug assert, with no else.